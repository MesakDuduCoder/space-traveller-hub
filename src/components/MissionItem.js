import React from 'react';
import Button from 'react-bootstrap/Button';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { cancelMission, joinMission } from '../redux/mission/missionSlice';

const MissionItem = ({
  id, name, description, reserved,
}) => {
  const dispatch = useDispatch();
  const member = reserved === true
    ? {
      variant: 'outline-danger', buttonName: 'leave mission', memberStatus: 'Active Member', memberBg: 'description-item blue',
    }
    : {
      variant: 'outline-secondary', buttonName: 'join mission', memberStatus: 'Not A Member', memberBg: 'description-item',
    };
  const clickHandler = (id) => (
    reserved === true ? dispatch(cancelMission(id)) : dispatch((joinMission(id))));
  return (
    <>
      <td>{name}</td>
      <td>{description}</td>
      <td className="flex">
        <p data-testid="activeMember" className={member.memberBg}>
          {member.memberStatus}
        </p>
      </td>
      <td className="flex">
        <Button
          data-testid="joinMission"
          variant={member.variant}
          onClick={() => clickHandler(id)}
          className="btn"
        >
          {member.buttonName}
        </Button>
      </td>
    </>
  );
};

MissionItem.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  id: PropTypes.string,
  reserved: PropTypes.bool,
};
MissionItem.defaultProps = {
  name: null,
  description: null,
  id: null,
  reserved: false,
};

export default MissionItem;
