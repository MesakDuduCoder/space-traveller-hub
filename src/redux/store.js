import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketSlice';
import missionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
    rockets: rocketsReducer,
  },
});

export default store;
