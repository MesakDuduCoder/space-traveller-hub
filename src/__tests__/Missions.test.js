import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MissionItem from '../components/MissionItem';
import store from '../redux/store';

describe('User Interactions test', () => {
  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionItem />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
