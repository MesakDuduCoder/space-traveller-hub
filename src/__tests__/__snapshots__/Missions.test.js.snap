// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
Array [
  <td />,
  <td />,
  <td
    className="flex"
  >
    <p
      className="description-item"
      data-testid="activeMember"
    >
      Not A Member
    </p>
  </td>,
  <td
    className="flex"
  >
    <button
      className="btn btn btn-outline-secondary"
      data-testid="joinMission"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      join mission
    </button>
  </td>,
]
`;
